### PR TITLE
build tweaks plus others

### PR DESCRIPTION
**Pull request by pietvanzoen**
- figured out that we could use the `Host` header to infer the api url so we don't need to configure `PLSFRSURL`. 
- renamed `lib` to `server`. seemed more consistent
- setup a `demo` folder
- updating readme
- other tweaks

it's toooo laaaate 😭

### CODE REVIEW

File: scripts/widget.sh [hunk 1]
```diff
@@ -1,8 +1,9 @@
-#!/bin/bash
+#!/bin/sh
```
Comment (pietvanzoen) [inline scripts/widget.sh:1]: didn't mean to commit this

File: README.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
 
 +:fries: as a service
 
+## Usage
+
+Add the following to a page where you would like to receive fries.
+
+```html
+<a id="plusfries" class="loading">🍟</a>
+<script src="//<your-plusfries-server>/plusfries.js"></script>
```
Comment (keepitterron) [inline README.md]: should we at least add an `async`?
i made the loader just to be a good 3rd party scripts citizen.

it's a tiny widget tho, I'm ok with straight loading it

File: demo/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# plusfries demo
```
Comment (keepitterron) [inline demo/README.md:1]: 🎉

File: server/embed.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+const script = fs.readFileSync("./public/plusfries.js").toString();
+
+module.exports = {
+  getEmbedScript(host) {
+    if (cache[host]) {
+      log.debug("Using script cache for " + host);
+      return cache[host];
+    }
+    log.debug("Building script for " + host);
+    cache[host] = script.replace("%PLUSFRIES_HOST%", host);
```
Comment (keepitterron) [inline server/embed.js:15]: 💃

**Review by keepitterron — APPROVED** [review]
✨ ✨ ✨ ✨ 💖 👍 WUNDERBAR! 👍 💖 ✨ ✨ ✨ ✨

**Comment by keepitterron** [comment]
(we use giphy at loom to accept PRs but they recently changed to branded pages instead of gifs so most of the times it breaks. I now use that wunderbar reply. WHAT ARE YOU GONNA DO ABOUT IT)

File: README.md (hunk 2 above)
Comment (pietvanzoen) [reply README.md]: yes! we should be good 3rd party citizens 😇

**Comment by pietvanzoen** [comment]
hahaha... YES! Let it spread! 

:sparkles: :sparkles: :sparkles: :sparkles: :sparkling_heart: :+1: WUNDERBAR! :+1: :sparkling_heart: :sparkles: :sparkles: :sparkles: :sparkles:

**Pull request merged**